File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionUpperTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionUpperTest extends BaseTest {
    @Test
    void uppercase() {
        var strVal = "text";
        assertTransformation(strVal, "$$upper:$", strVal.toUpperCase());
        assertTransformation(strVal, "$$upper():$", strVal.toUpperCase());
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionUpper.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";

describe("TransformerFunctionUpper", () => {
  test("uppercase", async () => {
    const strVal = "text";
    await assertTransformation(strVal, "$$upper:$", strVal.toUpperCase());
    await assertTransformation(strVal, "$$upper():$", strVal.toUpperCase());
  });
});
